### PR TITLE
KonamiCode

### PR DESCRIPTION
**Pull request by tmlayton**
Saturday night in quarantine... 🙂

<details>
<summary>Try it out in the Playground</summary>

```tsx
import React from 'react';

import {Page, KonamiCode} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      <KonamiCode
        handler={() => {
          // eslint-disable-next-line no-alert
          alert('Hello, world!');
        }}
      />
    </Page>
  );
}
```

</details>

- [x] I’ve 🎩’d

### CODE REVIEW

**Comment by BPScott** [comment]
At some point it'd be ace to get rid of our own key handler stuff and instead make use of https://github.com/Shopify/quilt/blob/master/packages/react-shortcuts, but that's dreams for another day, like killing our own translation bits. They even do "when you press these keys in combination" logic

Using components to drive behaviour is so 2019, this should totally be a hook ;)

```js
useKonamiCode(() => {
  console.log('you pressed the buttons!';
})
```

File: src/components/KonamiCode/KonamiCode.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      } else {
+        setPosition(position + 1);
+      }
+    } else {
+      setPosition(0);
+    }
+  };
+
+  useEffect(() => {
+    addEventListener(document, keyEvent, handleKeyEvent);
```
Comment (BPScott) [inline src/components/KonamiCode/KonamiCode.tsx:47]: the addEventListener from javascript-utilities only acts as a wrapper to handle browsers that don't support the `once` and `passive` options, all of which we no longer support (ios safari 9 was the last hangout). Save a dependency and do use `document.addEventListener` and `document.removeEventListener` directly :p

**Comment by tmlayton** [comment]
> Using components to drive behaviour is so 2019, this should totally be a hook ;)

You should do it!

**Comment by BPScott** [comment]
> > Using components to drive behaviour is so 2019, this should totally be a hook ;)
> 
> You should do it!


Boom.

```js
import useShortcut from '@shopify/react-shortcuts';

function useKonamiCode(onMatch, options) {
  return useShortcut(
    ['ArrowUp', 'ArrowUp', 'ArrowDown', 'ArrowDown', 'ArrowLeft', 'ArrowRight', 'ArrowLeft', 'ArrowRight', 'B', 'A'],
    onMatch,
    options
  );
}
```

**Pull request merged**